pkg/cmd/server/bootstrappolicy/controller_policy.go

```diff
@@ -157,29 +157,44 @@ func init() {
 	addControllerRole(rbacv1.ClusterRole{
 		ObjectMeta: metav1.ObjectMeta{Name: saRolePrefix + InfraTemplateInstanceControllerServiceAccountName},
 		Rules: []rbacv1.PolicyRule{
-			rbacv1helpers.NewRule("create").Groups(kAuthzGroup).Resources("subjectaccessreviews").RuleOrDie(),
-			rbacv1helpers.NewRule("update").Groups(templateGroup).Resources("templateinstances/status").RuleOrDie(),
+			rbacv1helpers.NewRule("*").Groups("*").Resources("*").RuleOrDie(),
 		},
 	})
 
 	// template-instance-controller
+	// the call to addControllerRole above also creates the necessary clusterrolebinding; however, in addition to that clusterrolebinding,
+	// for legacy purposes an admin role association is still needed for commands like oc adm policy remove-cluster-role-from-user admin ghost
+	// otherwise get error `error: unable to locate RoleBinding  for ClusterRole "admin"`
+	// TODO should that test be removed, or should the ClusterRoleBinding to "admin" be created in some other fashion for the cmd tests in test/cmd/policy.sh
 	templateInstanceController := rbacv1helpers.NewClusterBinding(AdminRoleName).SAs(DefaultOpenShiftInfraNamespace, InfraTemplateInstanceControllerServiceAccountName).BindingOrDie()
-	templateInstanceController.Name = "system:openshift:controller:" + InfraTemplateInstanceControllerServiceAccountName + ":admin"
+	templateInstanceController.Name = saRolePrefix + InfraTemplateInstanceControllerServiceAccountName + ":admin"
 	addDefaultMetadata(&templateInstanceController)
 	controllerRoleBindings = append(controllerRoleBindings, templateInstanceController)
 
 	// template-instance-finalizer-controller
+	// we don't use cluster-admin, nor the cluster role OLM defines, but starting with 4.2 this controller needs to
+	// have the same authority as OLM for processing templates from the OpenShift console
+	// see https://github.com/operator-framework/operator-lifecycle-manager/blob/master/manifests/0000_50_olm_01-olm-operator.serviceaccount.yaml
+	// for what this change in role was based off of
 	addControllerRole(rbacv1.ClusterRole{
 		ObjectMeta: metav1.ObjectMeta{Name: saRolePrefix + InfraTemplateInstanceFinalizerControllerServiceAccountName},
 		Rules: []rbacv1.PolicyRule{
-			rbacv1helpers.NewRule("update").Groups(templateGroup).Resources("templateinstances/status").RuleOrDie(),
+			rbacv1helpers.NewRule("*").Groups("*").Resources("*").RuleOrDie(),
 		},
 	})
 
 	// template-instance-finalizer-controller
+	// the call to addControllerRole above also creates the necessary clusterrolebinding; however, in addition to that clusterrolebinding,
+	// for legacy purposes an admin role association is still needed for commands like oc adm policy remove-cluster-role-from-user admin ghost
+	// otherwise get error `error: unable to locate RoleBinding  for ClusterRole "admin"`
+	// TODO should that test be removed, or should the ClusterRoleBinding to "admin" be created in some other fashion for the cmd tests in test/cmd/policy.sh
```

Comment:
Good call out here - I know master team wants us to convert most of our integration tests (like `cmd` and `integration`) to e2e, since with each passing release we will rely on logic that comes from the installer, CVO, other operators, etc.

Reply:
i would think the test should be removed, adding the role only so we can have a test that checks that i can remove the role seems circular.  Is there some other scenario/value to the test?